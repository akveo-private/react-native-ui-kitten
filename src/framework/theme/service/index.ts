export {
  Provider,
  Consumer,
} from './reactContext.service';
export { forwardProps } from './reactPropsForward.service';
