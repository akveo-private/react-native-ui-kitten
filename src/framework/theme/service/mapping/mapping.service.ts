import {
  ThemeMappingType,
  ComponentMappingType,
  AppearanceType,
  VariantGroupType,
  MappingType,
  StateType,
} from '../../component';

export const APPEARANCE_DEFAULT = 'default';

const FALLBACK_MAPPING_APPEARANCE = {};
const FALLBACK_MAPPING_VARIANT = {};

/**
 * @param component: string - component name
 * @param mapping: ThemeMappingType - theme mapping configuration object
 *
 * @return ComponentMappingType if presents in mapping, undefined otherwise
 */
export function getComponentMapping(mapping: ThemeMappingType,
                                    component: string): ComponentMappingType | undefined {

  return mapping[component];
}

export function getAppearance(mapping: ComponentMappingType,
                              appearance: string): AppearanceType | undefined {

  return mapping.appearance[appearance];
}

export function getAppearanceMapping(mapping: ComponentMappingType,
                                     appearance: string): MappingType | undefined {

  const appearanceConfig = getAppearance(mapping, appearance);

  return appearanceConfig && appearanceConfig.mapping;
}

export function getAppearanceMappingSafe(mapping: ComponentMappingType,
                                         appearance: string,
                                         fallback: MappingType): MappingType {

  return getAppearanceMapping(mapping, appearance) || fallback;
}

export function getStateAppearanceMapping(mapping: ComponentMappingType,
                                          appearance: string,
                                          state: string): any {

  const appearanceMapping = getAppearanceMappingSafe(mapping, appearance, FALLBACK_MAPPING_APPEARANCE);

  return getMappingState(appearanceMapping, state);
}

export function getStatelessAppearanceMapping(mapping: ComponentMappingType,
                                              appearance: string): any {

  const { state, ...params } = getAppearanceMappingSafe(mapping, appearance, FALLBACK_MAPPING_APPEARANCE);

  return params;
}

export function getAppearanceVariants(mapping: ComponentMappingType,
                                      appearance: string): VariantGroupType | undefined {

  const appearanceConfig = getAppearance(mapping, appearance);

  return appearanceConfig && appearanceConfig.variant;
}

export function getVariantMapping(mapping: ComponentMappingType,
                                  appearance: string,
                                  variant: string): MappingType | undefined {

  const variantGroupConfig = getAppearanceVariants(mapping, appearance);
  const variantGroupName = variantGroupConfig && Object.keys(variantGroupConfig).find(group => {
    return variantGroupConfig[group][variant] !== undefined;
  });
  const variantConfig = variantGroupName && variantGroupConfig[variantGroupName][variant];

  return variantConfig && variantConfig.mapping;
}

export function getVariantMappingSafe(mapping: ComponentMappingType,
                                      appearance: string,
                                      variant: string,
                                      fallback: MappingType): MappingType {

  return getVariantMapping(mapping, appearance, variant) || fallback;
}

export function getStateVariantMapping(mapping: ComponentMappingType,
                                       appearance: string,
                                       variant: string,
                                       state: string): any {

  const variantMapping = getVariantMappingSafe(mapping, appearance, variant, FALLBACK_MAPPING_VARIANT);

  return getMappingState(variantMapping, state);
}

export function getStatelessVariantMapping(mapping: ComponentMappingType,
                                           appearance: string,
                                           variant: string): any {

  const { state, ...params } = getVariantMappingSafe(mapping, appearance, variant, FALLBACK_MAPPING_VARIANT);

  return params;
}


export function getMappingState(mapping: MappingType,
                                state: string): StateType | undefined {

  return mapping.state && mapping.state[state];
}
