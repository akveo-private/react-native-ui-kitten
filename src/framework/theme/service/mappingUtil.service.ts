import {
  ThemeMappingType,
  ComponentMappingType,
  AppearanceType,
  VariantGroupType,
  MappingType,
  StateType,
} from '../component';

export const APPEARANCE_DEFAULT = 'default';

/**
 * @param component: string - component name
 * @param mapping: ThemeMappingType - theme mapping configuration object
 *
 * @return ComponentMappingType if presents in mapping, undefined otherwise
 */
export function getComponentMapping(mapping: ThemeMappingType,
                                    component: string): ComponentMappingType | undefined {

  return mapping[component];
}

export function getAppearance(mapping: ComponentMappingType,
                              appearance: string): AppearanceType | undefined {

  return mapping.appearance[appearance];
}

export function getAppearanceMapping(mapping: ComponentMappingType,
                                     appearance: string): MappingType | undefined {

  const appearanceConfig = getAppearance(mapping, appearance);

  return appearanceConfig && appearanceConfig.mapping;
}

export function getAppearanceMappingSafe(mapping: ComponentMappingType,
                                         appearance: string,
                                         fallback: MappingType): MappingType {

  return getAppearanceMapping(mapping, appearance) || fallback;
}

export function getAppearanceVariants(mapping: ComponentMappingType,
                                      appearance: string): VariantGroupType | undefined {

  const appearanceConfig = getAppearance(mapping, appearance);

  return appearanceConfig && appearanceConfig.variant;
}

export function getVariantMapping(mapping: ComponentMappingType,
                                  appearance: string,
                                  variant: string): MappingType | undefined {

  const variantGroupConfig = getAppearanceVariants(mapping, appearance);
  const variantGroupName = variantGroupConfig && Object.keys(variantGroupConfig).find(group => {
    return variantGroupConfig[group][variant] !== undefined;
  });
  const variantConfig = variantGroupName && variantGroupConfig[variantGroupName][variant];

  return variantConfig && variantConfig.mapping;
}

export function getVariantMappingSafe(mapping: ComponentMappingType,
                                      appearance: string,
                                      variant: string,
                                      fallback: MappingType): MappingType {

  return getVariantMapping(mapping, appearance, variant) || fallback;
}

export function getMappingState(mapping: MappingType,
                                state: string): StateType | undefined {

  return mapping.state && mapping.state[state];
}
