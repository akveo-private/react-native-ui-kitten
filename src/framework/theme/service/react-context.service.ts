import React from 'react';
import { ThemeShape } from '../component';

const defaultThemeValue: ThemeShape = {};

const {
  Provider,
  Consumer,
} = React.createContext(defaultThemeValue);

export {
  Provider,
  Consumer,
};
