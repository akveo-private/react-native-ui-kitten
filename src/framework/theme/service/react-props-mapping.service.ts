import React, { ComponentType } from 'react';

const REACT_METHODS = [
  'autobind',
  'childContextTypes',
  'componentDidMount',
  'componentDidUpdate',
  'componentWillMount',
  'componentWillReceiveProps',
  'componentWillUnmount',
  'componentWillUpdate',
  'contextTypes',
  'displayName',
  'getChildContext',
  'getDefaultProps',
  'getDOMNode',
  'getInitialState',
  'mixins',
  'propTypes',
  'render',
  'replaceProps',
  'setProps',
  'shouldComponentUpdate',
  'statics',
  'updateComponent',
];

export function forwardProps<P>(Source: ComponentType<P>, Target: ComponentType<P>): ComponentType<P> {
  function filterProps(prop) {
    // React specific methods and properties or properties from React's prototype
    const isReactProp = REACT_METHODS.includes(prop) || prop in React.Component.prototype;
    // Properties from enhanced component's prototype
    const isTargetProp = prop in Target.prototype;
    // Private methods
    const isPrivateProp = prop.startsWith('_');

    return !(isReactProp || isTargetProp || isPrivateProp);
  }

  function mapProps(prop) {
    if (typeof Source.prototype[prop] === 'function') {
      // Make sure the function is called with correct context
      Target.prototype[prop] = function (...args) {
        return Source.prototype[prop].apply(this.getConsumingComponent(), args);
      };
    } else {
      // Copy properties as getters and setters
      Object.defineProperty(Target.prototype, prop, {
        get() {
          return this.getConsumingComponent()[prop];
        },
        set(value) {
          this.getConsumingComponent()[prop] = value;
        },
      });
    }
  }

  Object.getOwnPropertyNames(Source.prototype).filter(filterProps).forEach(mapProps);

  return Target;
}
