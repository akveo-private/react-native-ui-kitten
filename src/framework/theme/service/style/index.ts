export * from './style.service';
export * from './styleConsumer.service';
