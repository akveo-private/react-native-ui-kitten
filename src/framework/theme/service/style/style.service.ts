import {
  APPEARANCE_DEFAULT,
  getStatelessAppearanceMapping,
  getStatelessVariantMapping,
  getStateAppearanceMapping,
  getStateVariantMapping,
} from '../mapping/mapping.service';
import {
  ComponentMappingType,
  ThemeType,
  StyleType,
} from '../../component';

const SEPARATOR_STATE = '.';

/**
 * Creates style object for variant/list of variants(optional) and its state/list of states(optional)
 *
 * Example
 *
 * appearance = 'outline';
 * variants = ['success', 'large'];
 * state = ['active', 'checked'];
 *
 * a = `default` + `outline`                    - acc appearance (apce) mapping
 *
 * v1 = `success` of `default`                  - `success` variant mapping of `default` apce
 * v2 = `success` of `outline`                  - `success` variant mapping of `outline` apce
 * v3 = `large` of `default`                    - `large` variant mapping of `default` apce
 * v4 = `large` of `outline`                    - `large` variant mapping of `outline` apce
 *
 * s1 = `active` of `default`                   - `active` state mapping of `default` apce
 * s2 = `active` of `outline`                   - `active` state mapping of `outline` apce
 * s3 = `active` of `default success`           - `active` state mapping of `success` variant of `default` apce
 * s4 = `active` of `outline success`           - `active` state mapping of `success` variant of `outline` apce
 * s5 = `active` of `default large`             - `active` state mapping of `large` variant of `default` apce
 * s6 = `active` of `outline large`             - `active` state mapping of `large` variant of `outline` apce
 *
 * s7 = `checked` of `default`                  - `checked` state mapping of `default` apce
 * s8 = `checked` of `outline`                  - `checked` state mapping of `outline` apce
 * s9 = `checked` of `default success`          - `checked` state mapping of `success` variant of `default` apce
 * s10 = `checked` of `outline success`         - `checked` state mapping of `success` variant of `outline` apce
 * s11 = `checked` of `default large`           - `checked` state mapping of `large` variant of `default` apce
 * s12 = `checked` of `outline large`           - `checked` state mapping of `large` variant of `outline` apce
 *
 * s13 = `active.checked` of `default`          - `active.checked` state mapping of `default` apce
 * s14 = `active.checked` of `outline`          - `active.checked` state mapping of `outline` apce
 * s15 = `active.checked` of `default success`  - `active.checked` state mapping of `success` variant of `default` apce
 * s16 = `active.checked` of `outline success`  - `active.checked` state mapping of `success` variant of `outline` apce
 * s17 = `active.checked` of `default large`    - `active.checked` state mapping of `large` variant of `default` apce
 * s18 = `active.checked` of `outline large`    - `active.checked` state mapping of `large` variant of `outline` apce
 *
 * res = a + (v1 + v2 + ... + vn) + (s1 + s2 + ... + sn)
 *
 * @param theme: ThemeType - theme object
 * @param mapping: ComponentMappingType - component theme mapping configuration
 * @param appearance: string - appearance applied to component
 * @param variants: string[] - variants applied to component. Default is []
 * @param states: string[] - states in which component is. Default is []
 *
 * @return StyleType - compiled component styles declared in mappings, mapped to theme values
 */
export function createStyle(theme: ThemeType,
                            mapping: ComponentMappingType,
                            appearance: string = APPEARANCE_DEFAULT,
                            variants: string[] = [],
                            states: string[] = []): StyleType {

  const normalizedAppearance = normalizeAppearance(appearance);
  const normalizedVariants = normalizeVariants(variants);
  const normalizedStates = normalizeStates(states);

  const appearanceMapping = reduce(normalizedAppearance, apce => {
    return getStatelessAppearanceMapping(mapping, apce);
  });

  const variantMapping = reduce(normalizedVariants, variant => {
    return reduce(normalizedAppearance, apce => {
      return getStatelessVariantMapping(mapping, apce, variant);
    });
  });

  const stateMapping = reduce(normalizedStates, state => {
    const appearanceStateMapping = reduce(normalizedAppearance, apce => {
      return getStateAppearanceMapping(mapping, apce, state);
    });

    const variantStateMapping = reduce(normalizedVariants, variant => {
      return reduce(normalizedAppearance, apce => {
        return getStateVariantMapping(mapping, apce, variant, state);
      });
    });

    return {...appearanceStateMapping, ...variantStateMapping};
  });

  return createStyleFromMapping({ ...appearanceMapping, ...variantMapping, ...stateMapping }, theme);
}

function createStyleFromMapping(mapping: any, theme: ThemeType): StyleType {
  return Object.keys(mapping).reduce((acc, current) => {
    const key = mapping[current];
    acc[current] = theme[key] || key;
    return acc;
  }, {});
}

function reduce(items: string[], next: (item: string) => any): any {
  return items.reduce((acc, current) => ({ ...acc, ...next(current) }), {});
}

/**
 * Creates normalized to design system array of component appearances
 *
 * Example:
 *
 * '' => ['default']
 * 'bold' => ['default', 'bold']
 * 'default' => ['default']
 * ...
 *
 * @param appearance: string - appearance applied to component
 *
 * @return string[] - array of merged appearances
 */
export function normalizeAppearance(appearance: string): string[] {
  return normalize([APPEARANCE_DEFAULT, appearance]);
}

/**
 * Creates normalized to design system array of component variants
 *
 * Example:
 *
 * [''] => []
 * ['success'] => ['success']
 * ['success', 'tiny'] => ['success', 'tiny']
 * ...
 *
 * @param variants: string[] - variants applied to component
 *
 * @return string[] - array of merged variants
 */
export function normalizeVariants(variants: string[]): string[] {
  return normalize(variants);
}

/**
 * Creates normalized to design system array of component states
 *
 * Example:
 *
 * [''] => []
 * ['active'] => ['active']
 * ['active', 'checked'] => ['active', 'checked', 'active.checked']
 * ['active', 'checked', 'disabled'] => ['active', 'checked', 'active.checked', 'disabled', 'active.checked.disabled']
 * ...
 *
 * @param states: string[] - states in which component is
 * @param separator - state separator. `.` in example
 *
 * @return string[] - array of merged states
 */
export function normalizeStates(states: string[], separator = SEPARATOR_STATE): string[] {
  const preprocess = normalize(states);
  if (preprocess.length === 0) {
    return preprocess;
  } else {
    return createStateDescription(preprocess, separator, []);
  }
}

function createStateDescription(state: string[], separator: string, result: string[]): string[] {
  if (state.length === 1) {
    const last = state[state.length - 1];
    return [last, ...result];
  } else {
    const concat = state.reduce((acc, current) => acc.concat(separator, current));
    const last = state.pop();

    result.unshift(last, concat);

    return createStateDescription(state, separator, result);
  }
}

function normalize(params: string[]): string[] {
  return noNulls(noDuplicates(params));
}

function noDuplicates(params: string[]): string[] {
  return [...new Set(params)];
}

function noNulls(params: string[]): string[] {
  return params.filter(Boolean);
}
