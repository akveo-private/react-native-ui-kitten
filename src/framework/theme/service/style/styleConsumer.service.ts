import { ComponentMappingType } from '../../component/mapping';
import { Props } from '../../component/style/styleConsumer.component';
import {
  Interaction,
  State,
} from '../../component/style/type';
import {
  APPEARANCE_DEFAULT,
  getAppearanceMappingSafe,
  getAppearanceVariants,
} from '../mapping/mapping.service';

export class StyleConsumerService {

  public getVariantPropKeys<P extends Props>(mapping: ComponentMappingType, props: P): string[] {
    const variants = getAppearanceVariants(mapping, APPEARANCE_DEFAULT);
    if (variants === undefined) {
      return [];
    }
    return Object.keys(props)
      .filter((key: string) => variants[key])
      .map((key: string) => props[key]);
  }

  public getStatePropKeys<P extends Props>(mapping: ComponentMappingType,
                                           props: P,
                                           interaction: Interaction[] = []): string[] {

    const appearance = getAppearanceMappingSafe(mapping, APPEARANCE_DEFAULT, {});

    const states = Object.keys(props)
      .filter((key: string) => appearance.state[key])
      .map(state => props[state] && State.parse(state))
      .filter(Boolean);

    return [...interaction, ...states];
  }
}
