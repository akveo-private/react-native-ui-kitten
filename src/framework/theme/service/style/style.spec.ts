import { APPEARANCE_DEFAULT } from '../mapping';
import {
  mapping,
  theme,
} from './style.spec.config';
import * as Service from './style.service';

describe('@style: service methods checks', () => {

  const { Test: testMapping } = mapping;

  describe('* preprocess', () => {

    it('* normalizes appearance properly', () => {
      const implicitDefault = Service.normalizeAppearance('default');
      const custom = Service.normalizeAppearance('custom');
      const empty = Service.normalizeAppearance('');
      const nullable = Service.normalizeAppearance(undefined);

      expect(implicitDefault).toEqual([
        'default',
      ]);
      expect(custom).toEqual([
        'default',
        'custom',
      ]);
      expect(empty).toEqual([
        'default',
      ]);
      expect(nullable).toEqual([
        'default',
      ]);
    });

    it('* normalizes variants properly', () => {
      const success = Service.normalizeVariants([
        'success',
      ]);
      const successTiny = Service.normalizeVariants([
        'success',
        'tiny',
      ]);
      const withDuplicates = Service.normalizeVariants([
        'success',
        'success',
        'tiny',
      ]);
      const withNulls = Service.normalizeVariants([
        'success',
        undefined,
        'tiny',
        null,
      ]);
      const empty = Service.normalizeVariants([
        '',
      ]);

      expect(success).toEqual([
        'success',
      ]);
      expect(successTiny).toEqual([
        'success',
        'tiny',
      ]);
      expect(withDuplicates).toEqual([
        'success',
        'tiny',
      ]);
      expect(withNulls).toEqual([
        'success',
        'tiny',
      ]);
      expect(withNulls).toEqual([
        'success',
        'tiny',
      ]);
      expect(empty).toEqual([]);
    });

    it('* normalizes states properly', () => {
      const active = Service.normalizeStates([
        'active',
      ]);
      const activeChecked = Service.normalizeStates([
        'active',
        'checked',
      ]);
      const activeCheckedDisabled = Service.normalizeStates([
        'active',
        'checked',
        'disabled',
      ]);
      const withDuplicates = Service.normalizeStates([
        'active',
        'checked',
        'active',
      ]);
      const withNulls = Service.normalizeStates([
        'active',
        undefined,
        'checked',
        null,
      ]);
      const empty = Service.normalizeStates([
        '',
      ]);
      const customSeparator = Service.normalizeStates([
        'active',
        'checked',
      ], '-');

      expect(active).toEqual([
        'active',
      ]);
      expect(activeChecked).toEqual([
        'active',
        'checked',
        'active.checked',
      ]);
      expect(activeCheckedDisabled).toEqual([
        'active',
        'checked',
        'active.checked',
        'disabled',
        'active.checked.disabled',
      ]);
      expect(withDuplicates).toEqual([
        'active',
        'checked',
        'active.checked',
      ]);
      expect(withNulls).toEqual([
        'active',
        'checked',
        'active.checked',
      ]);
      expect(empty).toEqual([]);
      expect(customSeparator).toEqual([
        'active',
        'checked',
        'active-checked',
      ]);
    });

  });

  describe('* styling', () => {

    describe('* default appearance', () => {

      it('* no variant and no state', () => {
        const style = Service.createStyle(theme, testMapping);
        expect(style).toMatchSnapshot();
      });

      describe('* with state', () => {

        it('* single', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            APPEARANCE_DEFAULT,
            [],
            ['active'],
          );

          expect(style).toMatchSnapshot();
        });

        it('* multiple', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            APPEARANCE_DEFAULT,
            [],
            ['active', 'checked'],
          );

          expect(style).toMatchSnapshot();
        });
      });

      describe('* with variant', () => {

        describe('* single', () => {

          it('* no state', () => {
            const style = Service.createStyle(
              theme,
              testMapping,
              APPEARANCE_DEFAULT,
              ['success'],
            );

            expect(style).toMatchSnapshot();
          });

          describe('* with state', () => {

            it('* single implicit (should apply from appearance)', () => {
              const style = Service.createStyle(
                theme,
                testMapping,
                APPEARANCE_DEFAULT,
                ['success'],
                ['active'],
              );

              expect(style).toMatchSnapshot();
            });

            it('* single explicit (should apply own)', () => {
              const style = Service.createStyle(
                theme,
                testMapping,
                APPEARANCE_DEFAULT,
                ['success'],
                ['checked'],
              );

              expect(style).toMatchSnapshot();
            });

            it('* multiple', () => {
              const style = Service.createStyle(
                theme,
                testMapping,
                APPEARANCE_DEFAULT,
                ['success'],
                ['active', 'checked'],
              );

              expect(style).toMatchSnapshot();
            });

          });

        });

        describe('* multiple', () => {

          it('* no state', () => {
            const style = Service.createStyle(
              theme,
              testMapping,
              APPEARANCE_DEFAULT,
              ['success', 'big'],
            );

            expect(style).toMatchSnapshot();
          });

          describe('* with state', () => {

            it('* single', () => {
              const style = Service.createStyle(
                theme,
                testMapping,
                APPEARANCE_DEFAULT,
                ['success', 'big'],
                ['active'],
              );

              expect(style).toMatchSnapshot();
            });

            it('* multiple', () => {
              const style = Service.createStyle(
                theme,
                testMapping,
                APPEARANCE_DEFAULT,
                ['success', 'big'],
                ['active', 'checked'],
              );

              expect(style).toMatchSnapshot();
            });

          });

        });

      });

    });

    describe('* custom appearance', () => {

      it('* no variant and no state', () => {
        const style = Service.createStyle(theme, testMapping, 'custom');

        expect(style).toMatchSnapshot();
      });

      describe('* with state', () => {

        it('* implicit (should apply from default appearance)', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            'custom',
            [],
            ['checked'],
          );

          expect(style).toMatchSnapshot();
        });

        it('* explicit (should apply own)', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            'custom',
            [],
            ['active'],
          );

          expect(style).toMatchSnapshot();
        });

      });

      describe('* with variant', () => {

        it('* implicit (should apply from default appearance)', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            'custom',
            ['big'],
          );

          expect(style).toMatchSnapshot();
        });

        it('* explicit (should apply own)', () => {
          const style = Service.createStyle(
            theme,
            testMapping,
            'custom',
            ['success'],
          );

          expect(style).toMatchSnapshot();
        });

      });

    });

    describe('* undefined appearance', () => {

      it('* no variant and no state (should apply default appearance)', () => {
        const style = Service.createStyle(theme, testMapping, 'undefined');

        expect(style).toMatchSnapshot();
      });

    });

  });

});
