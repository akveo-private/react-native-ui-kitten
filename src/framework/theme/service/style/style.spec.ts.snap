// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@style: service methods checks * styling * custom appearance * no variant and no state 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 4,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * custom appearance * with state * explicit (should apply own) 1`] = `
Object {
  "borderColor": "#E0E0E0",
  "borderWidth": 4,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * custom appearance * with state * implicit (should apply from default appearance) 1`] = `
Object {
  "borderColor": "#2196F3",
  "borderWidth": 4,
  "innerSize": 24,
  "selectColor": "#2196F3",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * custom appearance * with variant * explicit (should apply own) 1`] = `
Object {
  "borderColor": "#009688",
  "borderWidth": 4,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * custom appearance * with variant * implicit (should apply from default appearance) 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 4,
  "innerSize": 28,
  "selectColor": "transparent",
  "size": 42,
}
`;

exports[`@style: service methods checks * styling * default appearance * no variant and no state 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with state * multiple 1`] = `
Object {
  "borderColor": "#1976D2",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "#2196F3",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with state * single 1`] = `
Object {
  "borderColor": "#616161",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * multiple * no state 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 2,
  "innerSize": 28,
  "selectColor": "transparent",
  "size": 42,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * multiple * with state * multiple 1`] = `
Object {
  "borderColor": "#00796B",
  "borderWidth": 2,
  "innerSize": 28,
  "selectColor": "#009688",
  "size": 42,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * multiple * with state * single 1`] = `
Object {
  "borderColor": "#616161",
  "borderWidth": 2,
  "innerSize": 28,
  "selectColor": "transparent",
  "size": 42,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * single * no state 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * single * with state * multiple 1`] = `
Object {
  "borderColor": "#00796B",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "#009688",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * single * with state * single explicit (should apply own) 1`] = `
Object {
  "borderColor": "#009688",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "#009688",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * default appearance * with variant * single * with state * single implicit (should apply from appearance) 1`] = `
Object {
  "borderColor": "#616161",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;

exports[`@style: service methods checks * styling * undefined appearance * no variant and no state (should apply default appearance) 1`] = `
Object {
  "borderColor": "#9E9E9E",
  "borderWidth": 2,
  "innerSize": 24,
  "selectColor": "transparent",
  "size": 36,
}
`;
